src/Azure.php

```diff
@@ -129,6 +129,14 @@ protected function redirect(Request $request)
      */
     public function azurecallback(Request $request)
     {
+        // Added by smitthhyy 16Dec2019 - Return 403 if user authenticates in AD but is not assigned to this application
+        if ($request->isMethod('get')) {
+            $errorDescription = trim(substr($request->query('error_description', 'SOMETHING_ELSE'), 0, 11));
+            if($errorDescription == "AADSTS50105") {
+                abort(403, "User is not authorisied within Azure AD to access this application.");
```

Comment:
We should  `return abort(...` here.

Reply:
Let me try Daniel.

Reply:
New pull request.
README.md

To avoid an issue when following the Extended installation you get the following error: 
GuzzleHttp\Exception\ClientException Client error: `GET https://graph.microsoft.com/v1.0/me` resulted in a `401 Unauthorized` response: { "error": { "code": "InvalidAuthenticationToken", "message": "Access token validation failure. Invalid aud (truncated...) 

The cause of the error is incorrect AZURE_RESOURCE in the .env file which I had as https://graph.windows.net (per issue #10). It should be AZURE_RESOURCE="https://graph.microsoft.com" which I got from a cryptic clue at https://stackoverflow.com/questions/37885476/ms-graph-api-invalid-authentication-token plus it worked. 

Azure.php
Yes it works in fail() so moved it there. I will point out need to check Request $request as RequestException $e returns an unrelated error that is what drove the change. The error in $e is:
{"error":"invalid_request","error_description":"AADSTS900144: The request body must contain the following parameter: 'co (truncated...)

Reply:
> I will point out need to check Request $request as RequestException $e returns an unrelated error that is what drove the change.
Could you elaborate on what this means?

Reply:
The error message that is displayed when a user is username/password authenticated but doesn't have permission to run the application (prior to this change) is {"error":"invalid_request","error_description":"AADSTS900144: The request body must contain the following parameter: 'co (truncated...)
which is passed to the fail() function in RequestException $e. The meaningful message (and code) is passed to the fail() function in Request $request.  The original fail() function displays $e error which in this case is not relevant to the cause of the error.
The error message in $request is: {"error": "interaction_required","error_description": "AADSTS50105: The signed in user '{EmailHidden}' is not assigned to a role for the application 'uuid_here'(app_name_here)"}
Hope that help :)